odpc.client/src/composables/use-form-validator.ts

```diff
@@ -0,0 +1,100 @@
+import { ref, onUnmounted, type ComputedRef, watchEffect } from "vue";
+
+export const useFormValidator = (formRef: ComputedRef<HTMLFormElement | undefined>) => {
+  const isValid = ref(false);
+
+  const validateForm = () => {
+    if (!formRef.value) {
+      isValid.value = false;
+      return;
+    }
+
+    isValid.value = validateCheckboxGroups() && formRef.value.checkValidity();
+
+    if (!isValid.value) {
+      focusInvalidInput(formRef.value);
+    }
+  };
+
+  const validateCheckboxGroups = () => {
+    const checkboxGroups = getCheckboxGroups();
+
+    if (checkboxGroups.length === 0) return true;
+
+    return Array.from(checkboxGroups).every((group) => {
+      group.classList.remove("invalid");
+
+      const checkboxes = getCheckboxesFromGroup(group);
+
+      if (!isAnyChecked(checkboxes)) {
+        group.classList.add("invalid");
+        group.open = true;
+
+        checkboxes[0]?.focus();
+
+        return false;
+      }
+
+      return true;
+    });
+  };
+
+  // Input type text
+  const focusInvalidInput = (form: HTMLFormElement) => {
+    const invalidInputs = form.querySelectorAll(
+      "input[type='text']:invalid"
+    ) as NodeListOf<HTMLInputElement>;
+
+    invalidInputs.forEach((input) => {
+      input.classList.add("invalid");
+
+      const details = input.closest("details");
+      if (details) details.open = true;
+    });
+
+    invalidInputs[0]?.focus();
+  };
+
+  // Input type checkbox
+  const getCheckboxGroups = () =>
+    formRef.value?.querySelectorAll(
+      "details[data-required-group='checkbox']"
```

Comment:
misschien hoeven we hier niet af te dwingen dat het een details element is?

Reply:
Aangepast